### ISSUE TITLE
mesos-slave cannot connect to mesos-master

### ISSUE THREAD

**Issue by loshkovskyi** [CiscoCloud/mantl#991]
I encountered this issue on OpenStack with Mantl deployed from master branch. All `mesos-slave` services are not able to connect to the master ones:
>
Jan 13 12:36:24 mi-worker-001.cisco.com mesos-slave[8149]: 2016-01-13 12:36:24,084:8149(0x7f2700abf700):ZOO_ERROR@handle_socket_error_msg@1697: Socket [127.0.0.1:2181] zk retcode=-4, errno=111(Connection refused): server 
refused to accept the client

You'll have to change hostname of each worker node for the `mesos-slave` services to connect properly:
```
hostnamectl set-hostname mi-worker-001.novalocal
systemctl restart mesos-agent
```
Yes, I rebooted my nodes after upgrading packages as recommended here: https://microservices-infrastructure.readthedocs.org/en/latest/getting_started/index.html#provisioning-cloud-hosts (see the warning), so all hostnames were changed to the `cisco.com` domain.

This issue is related to the old one https://github.com/CiscoCloud/microservices-infrastructure/issues/27 and partially to https://github.com/CiscoCloud/microservices-infrastructure/issues/875.

So, I propose to fix it permanently and I see 2 solutions here:  

1) With `cloud-init`.
- Check that `cloud-init` is installed (it's already provided with CentOS image in most cases).
- Extend our OpenStack Terraform configuration with the relevant `cloud-init` template.

2) Without `cloud-init`.
- Check that `cloud-init` is absent (otherwise, it can silently alter your configuration on reboot).
- Extend our Ansible playbooks with a task to perform `hostnamectl`.

I've tested both solutions and they work well: the hostname is set correctly on reboots. I'd personally stay with the first option, as it's highly configurable and will allow to leverage `cloud-init` in the future.

**Comment by loshkovskyi** [comment]
Submitted PR https://github.com/CiscoCloud/microservices-infrastructure/pull/994.

Here is how it works after applying the Terraform configuration:
```
[root@mi-control-01 ~]# hostnamectl --static
mi-control-01.novalocal

[root@mi-control-01 ~]# cat /etc/hostname
mi-control-01.novalocal

[andrii@localhost]$ ansible-playbook -i plugins/inventory/terraform.py -e 'serial=0' playbooks/upgrade-packages.yml

[andrii@localhost]$ ansible all -i plugins/inventory/terraform.py -m shell -a "hostnamectl --static"
mi-control-03.novalocal
mi-worker-003.novalocal
mi-worker-001.novalocal
mi-control-01.novalocal
mi-control-02.novalocal
mi-edge-01.novalocal
mi-worker-002.novalocal
mi-edge-02.novalocal
```

A few more notes:
- Image must be supplied with `cloud-init` package since we use it in Terraform.
- I'm a bit worried that we manage `/etc/hostname` with Terraform (with this PR) and `/etc/hosts` with Ansible (the role `common`). We could manage both files with `cloud-init`.

**Comment by siddharthist** [comment]
Is this fixed by #1374?

**Comment by siddharthist** [comment]
Closed in #1374
